src/main/java/org/stellar/sdk/SorobanServer.java

```diff
@@ -428,6 +417,15 @@ private Transaction assembleTransaction(
     if (operation instanceof InvokeHostFunctionOperation) {
       // If the operation is an InvokeHostFunctionOperation, we need to update the auth entries if
       // existing entries are empty and the simulation result contains auth entries.
+      if (simulateTransactionResponse.getResults() == null
```

Comment:
@overcat , there was a fair amount of change done on the js client to handle the different types of simulation response payloads with different elements, as it could be for invoke, restore, and success/error, for reference the js client captured it like:
https://github.com/stellar/js-soroban-client/blob/main/src/soroban_rpc.ts#L168-L176

https://github.com/stellar/js-soroban-client/blob/main/src/transaction.ts#L111

Does the java parsing here need to be updated to same, it appears like so?

Reply:
I think it would be better if we annotate these through documents. The JS client has done more parsing of the response, but we haven't, we keep the original response.

Reply:
yes, also how 'success' is parsed from the raw simulation response from rpc is different now depending on whether it's indicating a restore is need as part of state exipration, and there's the optional present of the preamble member:
https://github.com/stellar/js-soroban-client/blob/main/src/soroban_rpc.ts#L226-L231

so, I think you're suggestion for documents, means stating these json rpc aspects for now in comments for sdk users to implement their own parsing, and then implement parsing internally in the sdk for the simulation response variations in a later iteration here?

Reply:
I have no intention of modifying it at the moment, as I mentioned earlier. Personally, I believe this would make things more complicated. If possible, let's keep it for now and listen to user feedback. We still have a chance to modify it before releasing the final version.

The link here contains detailed documents. Do you think I need to copy them into each field?
https://github.com/stellar/java-stellar-sdk/pull/530/files#diff-04e963815abf17aa3afc8545dd7431ba59aa861e57e3a04f53c5c33b82482305R12

https://soroban.stellar.org/api/methods/simulateTransaction#returns

Reply:
no, don't add docs on each field in SimulateTransactionResponse, I think the link to the soroban public docs is great to avoid duplicating docs here in code, and should be the source of record, but maybe we need to alert the sdk user why reading that link is needed due to the dynamic nature of the simulation response payload, how about this for the class docs on `SimulateTransactionResponse`:

```
/**
 * Response for JSON-RPC method simulateTransaction.
 *
 *  Note - The simulation response will have different model representations with different data memmbers
 *  present or absent depending representing the type of response that it is conveying. For example, the response for
 *  simulating an invoke host function, could be one of three types: error, success, or restore operation needed.
 *
 *  Please refer to the latest <a href="https://soroban.stellar.org/api/methods/simulateTransaction#returns"
 *  target="_blank">Soroban simulateTransaction documentation</a> for details on which data members of the
 *  simulation response model are keyed to each type of response.
 *
 */

```

Reply:
done - https://github.com/stellar/java-stellar-sdk/pull/530/commits/702628ca4d3438a688a5c917557baf5604c40c0b